### ISSUE TITLE
Is `inspect.getsourcelines` the best way to parse through the function for return values?

### ISSUE THREAD

**Issue by teald** [teald/porchlight#20]
Right now, `BaseDoor._get_return_vals` uses `inspect.getsourcelines` to retrieve the lines of code defining the function of interest, but this seems like a slow means of parsing for return values, especially if functions are large and/or `DynamicDoors` are considered (see issue #5). That's not to consider future features which may want to retrieve source multiple times, or in different ways.

If the source could be retrieved as a single string instead of a list, for example, then it could be parsed directly with `re`. Not sure how much of an improvement that would be.

Basically, this needs more thought/research. It works the way it is, and is not an issue, but might have a *much* better alternative.

**Comment by teald** [comment]
Thusfar I haven't found anything better, so I'm closing this issue for now. I'd say that, for v1.0, this will be kept pretty much intact barring significant bugs.
